community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py

```diff
@@ -320,6 +324,13 @@ def start_tpu(data):
 
 def resume_nodes(nodes: List[str], resume_data: Optional[ResumeData]):
     """resume nodes in nodelist"""
+    # Prevent inactive nodes associated with a future reservation from being resumed
+    active_nodes = nodes[:]
```

Comment:
This is bad way to filter list, both performance and readability wise.
Consider instead following:

```py
dormant_fr_nodes, nodes = util.separate(nodes, is_dormant_fr_node)

if dormant_fr_nodes:
  # do stuff here
```